ObjectAsset, ResourceAsset, and MaterialPaletteAsset

MoltonMontro opened the pull request:
- Added docs for ObjectAsset, ResourceAsset, and MaterialPaletteAsset.
- Added doc about Charts.unity3d file and the texture strips for it.
- Added mentions to objects in WeaponAsset doc.

## Questions/clarifications

- I've noted all of the ``Rubble`` object properties as being "deprecated in favor of ``Interactability``". Is this note accurate?
- Objects have an ``Interactability_Emission [flag]`` property, which seems to be used by vanilla light source objects (e.g., streetlights). I'm uncertain if this property is still implemented and functional, or if it's been removed/replaced. It's currently included in the docs, but if it's nonfunctional it should be removed.
- Just want to double-check that _all_ of the ``LOD`` object properties are related to interior culling.
- Resources have an ``Auto_Skybox [flag]`` property that I've described as "A skybox GameObject should automatically be made for this resource." Is this explanation accurate?
- Resources have a ``Radius [float]`` property, but I'm unsure if this is still functional after the changes to baking in the map editor. It's currently included in the docs, but if it's nonfunctional it should be removed.
- Is there a better name/title for the Charts.unity3d file documentation than the current "Charts" titling?
- Resources have a ``Scale [float]`` property that I've described as "The resource’s scale will be multiplied by a random number between 1 and this value.". Is this description is accurate?

MoltonMontro committed:
Document ObjectAsset and ResourceAsset

- Added docs for ObjectAsset, ResourceAsset, and MaterialPaletteAsset.
- Added doc about Charts.unity3d file and the texture strips for it.
- Clarified what BladeID affects in weapon docs.

MoltonMontro committed:
Fix typos/formatting

SDGNelson commented:
This is a huge update, thanks Molt!

Dang I typed up answers and then accidentally closed the tab. More concisely:

> I've noted all of the Rubble object properties as being "deprecated in favor of Interactability". Is this note accurate?

Initially, destruction was implemented as an interactable object type. Then it was separated into the Rubble properties to allow destruction of interactable objects. Both will always be supported, but the Rubble properties should be preferred for new objects because they are more specific.

> Objects have an Interactability_Emission [flag] property, which seems to be used by vanilla light source objects (e.g., streetlights). I'm uncertain if this property is still implemented and functional, or if it's been removed/replaced. It's currently included in the docs, but if it's nonfunctional it should be removed.

You're right: This should be removed! Will do.

> Just want to double-check that all of the LOD object properties are related to interior culling.

Yes, except for Decal_LOD_Bias.

> Resources have an Auto_Skybox [flag] property that I've described as "A skybox GameObject should automatically be made for this resource." Is this explanation accurate?

Mostly. A Skybox prefab should still be included with the appropriate mesh. This mesh has custom normals to match the lighting. For example, the pine trees have upward normals, whereas spherical trees have outward normals. The flag generates and assigns a material and texture automatically.

> Resources have a Radius [float] property, but I'm unsure if this is still functional after the changes to baking in the map editor. It's currently included in the docs, but if it's nonfunctional it should be removed.

Right again, I will remove!

> Is there a better name/title for the Charts.unity3d file documentation than the current "Charts" titling?

I think that's good for now. We can revisit if improving the in-game map.

> Resources have a Scale [float] property that I've described as "The resource’s scale will be multiplied by a random number between 1 and this value.". Is this description is accurate?

Ugh the implementation of this is a total hack. The final in-game scale is between 1.1 and 1.1 + scale * 2.

MoltonMontro committed:
Correct inaccuracies

Reworded info about Auto_Skybox, Scale, and Rubble. Removed documentation of since-nonexistent properties.

MoltonMontro commented:
Latest commit fixes stuff as per your answers. The PR should be merge ready.
